src/Avalonia.Base/Threading/DispatcherImplExceptionWrapper.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Threading;
+
+namespace Avalonia.Threading;
+
+internal class DispatcherImplExceptionWrapper : IControlledDispatcherImpl
+{
+    public DispatcherImplExceptionWrapper(IControlledDispatcherImpl inner)
+    {
+        _inner = inner;
+    }
+    
+    private readonly IControlledDispatcherImpl _inner;
+    public bool CurrentThreadIsLoopThread => _inner.CurrentThreadIsLoopThread;
+
+    public void Signal()
+    {
+        _inner.Signal();
+    }
+
+    public event Action? Signaled
+    {
+        add => _inner.Signaled += value;
+        remove => _inner.Signaled -= value;
+    }
+
+    public event Action? Timer
+    {
+        add => _inner.Timer += value;
+        remove => _inner.Timer -= value;
+    }
+
+    public long Now => _inner.Now;
+
+    public void UpdateTimer(long? dueTimeInMs)
+    {
+        _inner.UpdateTimer(dueTimeInMs);
+    }
+
+    public bool CanQueryPendingInput => _inner.CanQueryPendingInput;
+
+    public bool HasPendingInput => _inner.HasPendingInput;
+
+    public void RunLoop(CancellationToken token)
+    {
+        tryagainloop:
+        try
+        {
+            _inner.RunLoop(token);
+        }
+        catch (Exception e)
+        {
+            //RxApp.DefaultExceptionHandler.OnNext(e);
```

Comment:
This is a draft because I am not sure what is the best solution to propagate the exception. We can create another service to handle these. We can extend some common service (Application). etc. Any input is welcome. The goal is to have some code which would decide if the app should continue or not + have time to act.

Reply:
I would imagine the same old [Dispatcher.UnhandledException](http://msdn.microsoft.com/en-us/library/system.windows.threading.dispatcher.unhandledexception.aspx)